crates/bevy_pbr/src/wireframe.rs

```diff
@@ -53,31 +72,179 @@ fn extract_wireframe_config(mut commands: Commands, wireframe_config: Res<Wirefr
+            })
+        } else {
+            0
+        });
+        commands.entity(entity).insert(uniform_offset);
+    };
+
+    if config.global {
+        global_query.for_each(add_wireframe_uniform);
+    } else {
+        wireframe_query.for_each(add_wireframe_uniform);
+    }
+
+    wireframe_meta
+        .uniforms
+        .write_buffer(&render_device, &render_queue);
+}
+
+/// Internal [`WireframePlugin`] resource.  
+/// [`GlobalWireframeBindGroup`] stores the [`BindGroup`] of wireframe data that is used on the GPU side.
+struct GlobalWireframeBindGroup {
+    bind_group: BindGroup,
+}
+
+fn queue_wireframes_bind_group(
+    mut commands: Commands,
+    render_device: Res<RenderDevice>,
+    meta: Res<GlobalWireframeMeta>,
+    bind_group: Option<Res<GlobalWireframeBindGroup>>,
+) {
+    if bind_group.is_none() {
+        commands.insert_resource(GlobalWireframeBindGroup {
+            bind_group: render_device.create_bind_group(&BindGroupDescriptor {
+                entries: &[BindGroupEntry {
+                    binding: 0,
+                    resource: meta.uniforms.binding().unwrap(),
+                }],
+                label: Some("wireframe_bind_group"),
+                layout: &meta.bind_group_layout,
+            }),
+        });
     }
 }
 
-/// Controls whether an entity should rendered in wireframe-mode if the [`WireframePlugin`] is enabled
-#[derive(Component, Debug, Clone, Default, Reflect)]
+/// Controls per-entity wireframe rendering settings (if the [`WireframePlugin`] is enabled)
+#[derive(Component, Debug, Clone, Reflect)]
 #[reflect(Component)]
-pub struct Wireframe;
+pub struct Wireframe {
+    /// The color of this wireframe.  
+    /// For the [`Entity`] that has this [`Wireframe`] component, this color overrides any that was set in [`WireframeConfig::default_color`].
+    pub color: Color,
+}
 
-#[derive(Debug, Clone, Default)]
+impl Default for Wireframe {
+    fn default() -> Self {
+        Self {
+            color: Color::WHITE,
+        }
+    }
+}
+
+/// Configuration for [`WireframePlugin`].
+#[derive(Debug, Clone)]
 pub struct WireframeConfig {
-    /// Whether to show wireframes for all meshes. If `false`, only meshes with a [Wireframe] component will be rendered.
+    /// Whether to show wireframes for all meshes. If `false`, only meshes with a [`Wireframe`] component will be rendered.
     pub global: bool,
+    /// The default color for wireframes. If [`Self::global`] is set, any [`Entity`] that does not have a [`Wireframe`]
+    /// component attached to it will have wireframes in this color.
+    pub default_color: Color,
+}
+
+impl Default for WireframeConfig {
+    fn default() -> Self {
+        Self {
+            global: false,
+            default_color: Color::WHITE,
+        }
+    }
+}
+
+/// Internal [`WireframePlugin`] component.  
+/// [`WireframeUniformOffset`] holds the offset of a [`WireframeUniform`] in the [`GlobalWireframeMeta::uniforms`].
```

Comment:
```suggestion
/// Holds the offset of a [`WireframeUniform`] in the [`GlobalWireframeMeta::uniforms`].
```

In Rust, there's no need to repeat the name of the type like this in your doc comments.